vertical-pod-autoscaler/pkg/recommender/input/metrics/metrics_client.go

```diff
@@ -47,35 +44,29 @@ type ContainerMetricsSnapshot struct {
 type MetricsClient interface {
 	// GetContainersMetrics returns an array of ContainerMetricsSnapshots,
 	// representing resource usage for every running container in the cluster
-	GetContainersMetrics() ([]*ContainerMetricsSnapshot, error)
+	GetContainersMetrics(state *model.ClusterState) ([]*ContainerMetricsSnapshot, error)
 }
 
 type metricsClient struct {
-	metricsGetter resourceclient.PodMetricsesGetter
-	namespace     string
-	clientName    string
+	source     PodMetricsLister
+	namespace  string
+	clientName string
 }
 
 // NewMetricsClient creates new instance of MetricsClient, which is used by recommender.
 // namespace limits queries to particular namespace, use k8sapiv1.NamespaceAll to select all namespaces.
-func NewMetricsClient(config *rest.Config, namespace, clientName string) MetricsClient {
-	metricsGetter := resourceclient.NewForConfigOrDie(config)
-	return newMetricsClient(metricsGetter, namespace, clientName)
-}
-
-func newMetricsClient(metricsGetter resourceclient.PodMetricsesGetter, namespace, clientName string) MetricsClient {
+func NewMetricsClient(source PodMetricsLister, namespace, clientName string) MetricsClient {
 	return &metricsClient{
-		metricsGetter: metricsGetter,
-		namespace:     namespace,
-		clientName:    clientName,
+		source:     source,
+		namespace:  namespace,
+		clientName: clientName,
 	}
 }
 
-func (c *metricsClient) GetContainersMetrics() ([]*ContainerMetricsSnapshot, error) {
+func (c *metricsClient) GetContainersMetrics(state *model.ClusterState) ([]*ContainerMetricsSnapshot, error) {
```

Comment:
It looks like the `state` argument is not used, let's drop it

Reply:
Removed.

Reply:
Not yet. Maybe there was some problem with pushing the changes?